javageci-core/src/main/java/javax0/geci/annotationbuilder/AnnotationBuilder.java

```diff
@@ -0,0 +1,110 @@
+package javax0.geci.annotationbuilder;
+
+import javax0.geci.api.GeciException;
+import javax0.geci.api.Segment;
+import javax0.geci.api.Source;
+import javax0.geci.tools.AbstractJavaGenerator;
+import javax0.geci.tools.CompoundParams;
+import javax0.geci.tools.GeciReflectionTools;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Set;
+
+import static javax0.geci.api.Source.Set.set;
+import static javax0.geci.tools.CaseTools.ucase;
+
+public class AnnotationBuilder extends AbstractJavaGenerator {
+
+    private static class Config {
+        private String in = "";
+    }
+
+    @Override
+    public void process(Source source, Class<?> klass, CompoundParams global) throws Exception {
+        try {
+            final var mnemonic = getMnemonic(klass);
+            final var keys = getImplementedKeysSorted(klass);
+            final var annotation = ucase(mnemonic);
+            final var in = config.in;
+
+            final var newSource = source.newSource(set(in), annotation + ".java");
+            writeContent(newSource, mnemonic, annotation, keys);
+        } catch (NoSuchMethodException ex) {
+            throw new GeciException("Cannot generate annotation for " + klass.getName() + " because it does not have a mnemonic or implementedKeys method.");
+        }
+    }
+
+    private ArrayList<String> getImplementedKeysSorted(final Class<?> klass)
+            throws NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
+        final Method implementedKeysMethod = GeciReflectionTools.getMethod(klass, "implementedKeys");
```

Comment:
I do not like this variable name, because it is too long. There is no point in writing `Method` to the end. That is the type of the variable and today all IDEs just show that immediately. We do not use the Microsoft invented Hungarian notation to signal the type of the variable (which originates from FORTRAN). When I `setAccessible()` or `invoke()` I can see that this is a method.

At the same time, the `implementedKeys` part of the name is also redundant. There is no other `Method` object in the code and since the method, `getImplementedKeysSorted` is fairly short (as it should be) the name of the local variable can be something simpler without jeopardizing readability.

Reply:
Understood. Renamed the variables to have shorter but still meaningful names.